### PR TITLE
Maven wagon support

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/653)

### CODE REVIEW

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 1]
```diff
@@ -460,4 +470,31 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
...
+
+        val legacySupport = container.lookup(LegacySupport::class.java, "default")
+        legacySupport.session = mavenSession
+
+        val sessionScope = container.lookup(SessionScope::class.java, "default")
+        sessionScope.enter()
+
+        try {
+            sessionScope.seed(MavenSession::class.java, mavenSession)
+
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Nit: Maybe remove this blank line?

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 2]
```diff
@@ -460,4 +470,31 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
...
+            return block()
+        } finally {
+            sessionScope.exit()
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Note that this will get executed even if `return`ing successfully above,

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 3]
```diff
@@ -460,4 +470,31 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
 
         return VcsInfo(type, url, tag)
     }
+
+    /**
+     * Create a [MavenSession] and setup the [LegacySupport] and [SessionScope] because this is required to load
+     * extensions using Maven Wagon.
+     */
+    private fun <R> wrapMavenSession(block: () -> R): R{
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Nit: Missinf space before `{`.

File: analyzer/build.gradle [hunk 4]
```diff
@@ -28,6 +28,7 @@ dependencies {
     compile "org.apache.maven.resolver:maven-resolver-connector-basic:$mavenResolverVersion"
     compile "org.apache.maven.resolver:maven-resolver-transport-file:$mavenResolverVersion"
     compile "org.apache.maven.resolver:maven-resolver-transport-http:$mavenResolverVersion"
+    compile "org.apache.maven.resolver:maven-resolver-transport-wagon:$mavenResolverVersion"
```
Comment (sschuberth) [inline analyzer/build.gradle]: Does it really make sense to have a separate commit for this change, or should we squash?

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 3 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.
